src/lib.rs

```diff
@@ -170,28 +175,31 @@ mod tls {
                 |error| InternalConnectError::ReadFile { file: path.into(), error });
             let mut reader = &*contents;
 
-            let mut certs = try_map_err!(rustls_pemfile::certs(&mut reader),
+            let certs = try_map_err!(rustls_pemfile::certs(&mut reader),
                 |error| InternalConnectError::ParseCert { file: path.into(), error });
 
             if certs.len() != 1 {
-                return Err(InternalConnectError::InvalidCertCount { file: path.into(), count: certs.len(), });
+                tracing::debug!("Certificate count is not 1 (Actual Count: {})", certs.len());
```

Comment:
Unconditional debug with message "Loaded certificate chain with {} certificates" (or similar) would be better since we don't treat `1` as special.

Could you also remove `InvalidCertCount` from `InternalConnectError`? It's not used anymore.